airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py

```diff
@@ -601,34 +601,31 @@ def cleanup_stuck_queued_tasks(self, tis: list[TaskInstance]) -> list[str]:
         :param tis: List of Task Instances to clean up
         :return: List of readable task instances for a warning message
         """
-        from airflow.providers.cncf.kubernetes.pod_generator import PodGenerator
-
         if TYPE_CHECKING:
             assert self.kube_client
             assert self.kube_scheduler
-        readable_tis = []
+        readable_tis: list[str] = []
+        if not tis:
+            return readable_tis
+        pod_labels_combined_str_to_pod_map = self.get_pod_labels_combined_str_to_pod_map()
         for ti in tis:
-            selector = PodGenerator.build_selector_for_k8s_executor_pod(
-                dag_id=ti.dag_id,
-                task_id=ti.task_id,
-                try_number=ti.try_number,
```

Comment:
I think you didn't add the try_number selector

Reply:
I have removed that flag intentionally. I belive dag_id, task_id and run_id sufficient to identify the pods uniquely across the system. Do you know any use case where try_number differs?

Reply:
Based on my observations, the `run_id` remains the same across retries, even though the `try_number` changes.

Reply:
Yes. I believe one pod exists in the system with a pending state for the same run_id.  is it possible to have multiple pods with different try_numbers simultaneously?

Reply:
I think the answer is no

Reply:
If no, then try_number doesn't play a role here.

Reply:
Well I think it can be but in other status.
For example a pod that stuck in terminating and another one (with different try number) the started.
Is it matter?

Reply:
We are looking for pending state pods only. More than one pod doesn't exists in the same pending state. So, adding try_number doesn't  add any value here.